datafusion/expr/src/utils.rs

```diff
@@ -739,6 +739,35 @@ pub fn from_plan(
             Ok(plan.clone())
         }
         LogicalPlan::DescribeTable(_) => Ok(plan.clone()),
+        LogicalPlan::Unnest(Unnest { column, schema, .. }) => {
+            // Update schema with unnested column type.
+            let input = Arc::new(inputs[0].clone());
+            let nested_field = input.schema().field_from_column(column)?;
+            let unnested_field = schema.field_from_column(column)?;
+            let fields = input
+                .schema()
+                .fields()
+                .iter()
+                .map(|f| {
+                    if f == nested_field {
+                        unnested_field.clone()
+                    } else {
+                        f.clone()
+                    }
+                })
+                .collect::<Vec<_>>();
+
+            let schema = Arc::new(DFSchema::new_with_metadata(
+                fields,
+                input.schema().metadata().clone(),
+            )?);
+
+            Ok(LogicalPlan::Unnest(Unnest {
+                input,
+                column: column.clone(),
+                schema,
+            }))
+        }
```

Comment:
I think the intent of `from_plan` is to be mechanical construction rather than potentially changing the schema, so I was surprised this code is necessary.

However, when I tried changing this code to be something more like

```rust
            Ok(LogicalPlan::Unnest(Unnest {
                input: inputs[0].clone().into(),
                column: column.clone(),
                schema: schema.clone(),
            }))
```

The test fails:

```

---- unnest_columns stdout ----
Error: Internal("PhysicalExpr Column references column 'tags' at index 2 (zero-based) but input schema only has 1 columns: [\"tags\"]")


failures:
    unnest_columns
```

Reply:
Ah yes, I had to add this code because tests were failing due to a projection push-down when calling an aggregate function like count. At this point the child schema has only the count field while the unnest schema has all the fields in the initial projection.